### DISCUSSION TITLE
Importing from DLL works only when namespace != assembly name

### DISCUSSION THREAD

**Discussion by mardukbp** [pythonnet/pythonnet#2420] [Q&A] [answered]
### Environment

-   Pythonnet version: 3.0.3
-   Python version: 3.11.9
-   Operating System: 10.0.19045
-   .NET Runtime: 8.0.301

### Details

I compiled the following code to `CalcTest.dll` with target framework netstandard2.0:

```csharp
namespace CalcTest
{
    public static class Calculator
    {
        public static int Add(int a, int b)
        {
            return a + b;
        }
    }
}
```

Then I tried importing the class `Calculator` from Python:

```python
import sys
from pathlib import Path

sys.path.append(str(Path(__file__).parent / "DLLs"))

import clr

clr.AddReference("CalcTest")

from CalcTest import Calculator
```

and I obtained the following error:

```
ImportError: cannot import name 'Calculator' from 'CalcTest' (unknown location)
```

Following the tutorial [How To Call A Dynamic Library](https://github.com/pythonnet/pythonnet/wiki/How-to-call-a-dynamic-library) I changed the namespace in C# to `CalcTestNS`, recompiled the project and updated the import in Python:

```python
from CalcTestNS import Calculator
```

and it worked.

In the tutorial it is not mentioned that the namespace and the assembly cannot have the same name. It would be a great time-saver to document this.

**Comment by filmor** [comment, marked as answer]
It's not mentioned because it's not true. For Python scripts, the current path is implicitly added to `sys.path` and thus the directory `CalcTest` is loaded as a namespace package. Rename it and it will be loaded properly.

**Reply by mardukbp** [reply]
Thank you for the explanation! I checked sys.path and realized that the parent directory of the Python module is there. The naming conflict arises because the C# project is in this directory. The hint you gave me is not trivial. It should be added to the tutorial on importing dynamic libraries.
